src/core/elf.c

```diff
@@ -3,10 +3,54 @@
 #include <mmu/paging.h>
 #include <string.h>
 
+#define ELF_TYPE_EXECUTABLE         0x2
+
+#define ELF_PROGRAM_TYPE_LOAD       0x1
+
+#define ELF_FLAG_EXECUTE            0x1
+#define ELF_FLAG_READ               0x4
+#define ELF_FLAG_WRITE              0x2
+
+#define ELF_SECTION_INDEX_UNDEFINED 0 //Undefined or meaningless section reference
+
+#define ELF_SECTION_TYPE_NULL       0 //Unused section header
+#define ELF_SECTION_TYPE_PROGBITS   1 //Information defined by the program
+#define ELF_SECTION_TYPE_NOBITS     8 //Uninitialized Space, doesn't use any space in the file
+
+#define ELF_SECTION_FLAG_WRITE      0x1 //Contains writable data
+#define ELF_SECTION_FLAG_ALLOC      0x2 //Section is allocated in memory image of program
+#define ELF_SECTION_FLAG_EXECINSTR  0x4 //Contains executable instructions
+
+inline elf_section_header_t* elf_section_header(elf_header_t* header) {
```

Comment:
Can we remove the `inline` keyword? I get the idea but is it necessary? If I remember correctly, the compiler might not inline the function anyway.

Reply:
We could maybe add a `get_` prefix to these helper functions, WDYT? (I am not sure)

Reply:
Last but not least, can we declare the prototypes of these functions below the existing prototypes and move the definitions of these functions at the bottom of the file?

Reply:
Did a test in my machine:
# Without inline 
```
objdump -t build/libk-willOS.a | grep elf
...
0000000000000000 l    df *ABS*	0000000000000000 elf.c
0000000000000000 g     F .text	0000000000000122 is_elf
0000000000000220 g     F .text	0000000000000451 elf_load
0000000000000680 g     F .text	0000000000000008 get_elf_section_header
0000000000000690 g     F .text	000000000000000f get_elf_section
00000000000006a0 g     F .text	0000000000000023 elf_str_table
00000000000006d0 g     F .text	0000000000000033 elf_lookup_string
...
```

# With inline and inline_always attribute:
```objdump -t build/libk-willOS.a | grep elf
...
0000000000000000 l    df *ABS*	0000000000000000 elf.c
0000000000000000 g     F .text	0000000000000122 is_elf
0000000000000220 g     F .text	0000000000000451 elf_load
...
```

So, they were certainly inlined, and since they're very small functions we can save a lot of function calls which will speed up performance overall. Inline is supposed to be used for smaller functions either way, so I would suggest we keep inlining.